src/BugsnagUnity.mm

```diff
@@ -129,11 +129,273 @@
+
+    NSArray * theErrors = ((__bridge BugsnagEvent *)event).errors;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theErrors.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theErrors.count; i++) {
+       c_array[i] = (__bridge void *)theErrors[i];
+    }
+    callback(instance, c_array, [theErrors count]);
+    free(c_array);
+}
+
+void bugsnag_getBreadcrumbsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *breadcrumbs[], int breadcrumbs_size)) {
+
+    NSArray * theBreadcrumbs = ((__bridge BugsnagEvent *)event).breadcrumbs;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theBreadcrumbs.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theBreadcrumbs.count; i++) {
+       c_array[i] = (__bridge void *)theBreadcrumbs[i];
+    }
+    callback(instance, c_array, [theBreadcrumbs count]);
+    free(c_array);
+}
+
+void bugsnag_getBreadcrumbMetadata(const void *breadcrumb,const void *instance, void (*callback)(const void *instance,const char *keys[], int keys_size, const char *values[], int values_size)) {
+     NSDictionary* sectionDictionary = ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata;
+     NSArray *keys = [sectionDictionary allKeys];
+     NSArray *values = [sectionDictionary allValues];
+     int count = 0;
+     if ([keys count] <= INT_MAX) {
+       count = (int)[keys count];
+     }
+     const char **c_keys = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
+     const char **c_values = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
+     for (NSUInteger i = 0; i < (NSUInteger)count; i++) {
+       c_keys[i] = [[keys objectAtIndex: i] UTF8String];
+       c_values[i] = [[[values objectAtIndex: i]description] UTF8String];
+     }
+    callback(instance,c_keys, count, c_values, count);
+    free(c_keys);
+    free(c_values);
+}
+
+void bugsnag_setBreadcrumbMetadata(const void *breadcrumb, const char *metadata[], int metadataCount) {
+  NSMutableDictionary *ns_metadata = [NSMutableDictionary new];
+  for (int i = 0; i < metadataCount; i += 2) {
+    if (metadata[i] && metadata[i+1]) {
+        ns_metadata[@(metadata[i])] = @(metadata[i+1]);
+    }
+
+  }
+  ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata = ns_metadata;
+}
+
+const char * bugsnag_getBreadcrumbType(const void *breadcrumb){
+    return strdup([BSGBreadcrumbTypeValue(((__bridge BugsnagBreadcrumb *)breadcrumb).type) UTF8String]);    
+}
+
+void bugsnag_setBreadcrumbType(const void *breadcrumb, char * type){
+    ((__bridge BugsnagBreadcrumb *)breadcrumb).type = BSGBreadcrumbTypeFromString(@(type));
+}
+
+const char * bugsnag_getLongValue(const void *object, char * key){
+    id value = [(__bridge id)object valueForKey:@(key)];
+    if ([value isKindOfClass:[NSNumber class]])
+    {
+        return strdup([[value stringValue] UTF8String]);
+    }
+    else
+    {
+        return strdup("null");
```

Comment:
Can we just return `NULL` here, rather than a string containing `"null"`?

Reply:
Done 👍